src/parser/Tokenizer.cc

```diff
@@ -168,6 +168,17 @@ Parser::Tokenizer::skipOne(const CharacterSet &chars)
     return false;
 }
 
+char
+Parser::Tokenizer::skipOne()
+{
+    if (buf_.isEmpty())
+        throw InsufficientInput();
```

Comment:
This is not the method we discussed in https://github.com/squid-cache/squid/pull/1540#discussion_r1372563952. This method throws on EOF. The method we discussed returns std::nullopt in that case (otherwise, we would not be able to use it in an `if` statement as sketched).

If you want to keep this skipOne() method, please give it a `description` parameter -- all Tokenizer methods that throw have a description parameter. That is how we distinguish them from the non-throwing methods...

To implement the method we discussed, add another skipOne(void) method that returns std::nullopt on EOF. The throwing skipOne(description) should call skipOne(void).

I recommend doing both, but I cannot insist on it.